### ISSUE TITLE
bug(typescript): quick-lint crashes with unclosed TSX JavaScript expression-thingy `{`

### ISSUE THREAD

**Issue by vegerot** [quick-lint/quick-lint-js#1108]
Steps to reproduce:

1. write

```tsx
import { j } from "myImport";
type k = {
};
let i = (<div
  router=
>
  hi
</div>
)
```

2. on line 5, type a `{` at the end of the line, like

```tsx
import { j } from "myImport";
type k = {
};
let i = (<div
  router={
>
  hi
</div>
)
```

Expected: quick-lint should report an error for unclosed JSX JS expression-thingy

Actual: quick-lint crashes.

See this video for reproduction:

https://github.com/quick-lint/quick-lint-js/assets/5297725/78bd264d-4e97-4882-905c-0e0715717705

**Comment by henrikvilhelmberglund** [comment]
Also seeing this.
